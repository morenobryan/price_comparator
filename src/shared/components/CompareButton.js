// @flow
import * as React from 'react';
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import { secondaryTextColor, textColor } from '../../shared/styles';

type Props = {
  color: string,
  disabled: boolean,
  disabledColor: string,
  label: string,
  navigation: { navigate: string => void },
  nextPage: string,
  icon: string,
};

const CompareButton = ({
  color,
  disabled,
  disabledColor,
  icon,
  label,
  navigation,
  nextPage,
}: Props) => (
  <TouchableOpacity
    style={[
      styles.submitButton,
      disabled
        ? { borderColor: 'transparent', backgroundColor: disabledColor }
        : { borderColor: textColor, backgroundColor: color, elevation: 3 },
    ]}
    onPress={() => navigation.navigate(nextPage)}
    disabled={disabled}
  >
    <Text style={[disabled ? styles.disabledSubmitLabel : styles.submitLabel, styles.icon]}>
      {icon}
    </Text>
    <Text style={disabled ? styles.disabledSubmitLabel : styles.submitLabel}>{label}</Text>
  </TouchableOpacity>
);

export default CompareButton;

const styles = StyleSheet.create({
  submitLabel: {
    color: textColor,
    fontFamily: 'proximaNovaAltRegular',
  },
  disabledSubmitLabel: {
    color: '#eee',
    fontFamily: 'proximaNovaAltRegular',
  },
  submitButton: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    borderRadius: 30,
    padding: 20,
    marginLeft: 20,
    marginRight: 20,
    marginTop: 10,
    marginBottom: 10,
    borderWidth: 1,
  },
  icon: {
    paddingRight: 5,
    fontFamily: 'flaticon',
  },
});
