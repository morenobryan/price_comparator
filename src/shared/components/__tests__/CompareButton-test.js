import 'react-native';
import React from 'react';
import CompareButton from '../CompareButton';

import renderer from 'react-test-renderer';

const props = {
  color: '#fff',
  disabled: false,
  disabledColor: '#000',
  label: 'Test Label',
  navigation: { navigate: () => {} },
  nextPage: 'Dashboard',
  icon: 'Test Icon',
};

test('renders correctly', () => {
  const tree = renderer.create(<CompareButton {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
