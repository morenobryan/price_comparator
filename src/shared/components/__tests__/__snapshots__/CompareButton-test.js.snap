// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#3D3527",
      "borderRadius": 30,
      "borderWidth": 1,
      "elevation": 3,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 10,
      "marginLeft": 20,
      "marginRight": 20,
      "marginTop": 10,
      "opacity": 1,
      "padding": 20,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#3D3527",
          "fontFamily": "proximaNovaAltRegular",
        },
        Object {
          "fontFamily": "flaticon",
          "paddingRight": 5,
        },
      ]
    }
  >
    Test Icon
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#3D3527",
        "fontFamily": "proximaNovaAltRegular",
      }
    }
  >
    Test Label
  </Text>
</View>
`;
